app/src/main/java/com/cmput301f20t21/bookfriends/ui/library/owned/MapDialog.java

```diff
@@ -168,10 +169,13 @@ public void onClick(View view) {
         confirmSearchButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                // accept the request only when user specifies the location of meeting
-                Toast.makeText(context, getString(R.string.accepted_request), Toast.LENGTH_SHORT).show();
-                vm.acceptRequest(position);
-                dialog.dismiss();
+                if (meetingLocation != null) {
+                    // accept the request only when user specifies the location of meeting
+                    Toast.makeText(context, getString(R.string.accepted_request), Toast.LENGTH_SHORT).show();
+                    vm.acceptRequest(position, meetingLocation);
+                    dialog.dismiss();
+                }
+                Toast.makeText(context, getString(R.string.pick_location), Toast.LENGTH_SHORT).show();
```

Comment:
this Toast will show up even if there's a meeting location? The logic here seems like it also I don't see it in the UI

Reply:
Yeah my bad, the Toast should be in the condition when meetingLocation == null.